pkg/cmd/get/packages.go

```diff
@@ -0,0 +1,149 @@
+package get
+
+import (
+	"context"
+	"fmt"
+	"github.com/cnoe-io/idpbuilder/api/v1alpha1"
+	"github.com/cnoe-io/idpbuilder/pkg/build"
+	"github.com/cnoe-io/idpbuilder/pkg/k8s"
+	"github.com/cnoe-io/idpbuilder/pkg/printer"
+	"github.com/cnoe-io/idpbuilder/pkg/printer/types"
+	"github.com/cnoe-io/idpbuilder/pkg/util"
+	"github.com/spf13/cobra"
+	"io"
+	"k8s.io/client-go/util/homedir"
+	"os"
+	"path/filepath"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strconv"
+)
+
+var PackagesCmd = &cobra.Command{
+	Use:          "packages",
+	Short:        "retrieve packages from the cluster",
+	Long:         ``,
+	RunE:         getPackagesE,
+	SilenceUsage: true,
+}
+
+func getPackagesE(cmd *cobra.Command, args []string) error {
+	ctx, ctxCancel := context.WithCancel(cmd.Context())
+	defer ctxCancel()
+	kubeConfigPath := filepath.Join(homedir.HomeDir(), ".kube", "config")
+
+	opts := build.NewBuildOptions{
+		KubeConfigPath: kubeConfigPath,
+		Scheme:         k8s.GetScheme(),
+		CancelFunc:     ctxCancel,
+		TemplateData:   v1alpha1.BuildCustomizationSpec{},
+	}
+
+	b := build.NewBuild(opts)
+
+	kubeConfig, err := b.GetKubeConfig()
+	if err != nil {
+		return fmt.Errorf("getting kube config: %w", err)
+	}
+
+	kubeClient, err := b.GetKubeClient(kubeConfig)
+	if err != nil {
+		return fmt.Errorf("getting kube client: %w", err)
+	}
+
+	return printPackages(ctx, os.Stdout, kubeClient, outputFormat)
+}
+
+// Print all the custom packages or based on package arguments passed using flag: -p
+func printPackages(ctx context.Context, outWriter io.Writer, kubeClient client.Client, format string) error {
+	packageList := []types.Package{}
+	customPackages := v1alpha1.CustomPackageList{}
+	var err error
+
+	idpbuilderNamespace, err := getIDPNamespace(ctx, kubeClient)
+	if err != nil {
+		return fmt.Errorf("getting namespace: %w", err)
+	}
+
+	argocdBaseUrl := util.ArgocdBaseUrl()
+
+	if len(packages) == 0 {
+		// Get all custom packages
+		customPackages, err = getPackages(ctx, kubeClient, idpbuilderNamespace)
+		if err != nil {
+			return fmt.Errorf("listing custom packages: %w", err)
+		}
+	} else {
+		// Get the custom package using its name
+		customPackages = v1alpha1.CustomPackageList{}
```

Comment:
You can remove this since we created one in the outer scope.

Reply:
Fixed: fb6248d